hydromt/models/v1/grid.py

```diff
@@ -0,0 +1,749 @@
+"""Grid ModelComponent."""
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import xarray as xr
+from affine import Affine
+
+# TODO change import when moving this file
+from pyproj import CRS
+from shapely.geometry import box
+
+from hydromt import gis_utils, raster, workflows
+from hydromt.data_catalog import DataCatalog
+from hydromt.nodata import NoDataStrategy, _exec_nodata_strat
+from hydromt.typing import DeferedFileClose
+
+from .model_component import ModelComponent  # TODO change import when moving this file
+from .model_utils import (
+    _assert_read_mode,
+    _assert_write_mode,
+    read_nc,
+    write_nc,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GridModelComponent(ModelComponent):
+    """GridModelComponent class."""
+
+    def __init__(self, root, logger=logger):
+        # self._model_ref = weakref.ref(model) TODO: discuss if this is necessary
```

Comment:
We need to pass model rather than root as we also want to be able to access other properties such as region, datacatalog, etc. See also discussion on model components  https://github.com/Deltares/hydromt/discussions/636

Reply:
I think it would reduce complexity if we initialize a model component with model,root, model.region, model.datacatalog etc., instead of passing the model object. This way the model object and model component object are kept separate.

Reply:
In the discussion that's also where we started. However, we found that other properties of model might also be needed by some components and in the future new properties. Passing model with a weak_ref works for all and can therefore be implemented in a general way for all components and avoid potential breaking changes in future versions if new properties are required.